jobs/extract_content/worker.go

```diff
@@ -0,0 +1,176 @@
+			mlog.Debug("Worker received stop signal", mlog.String("worker", w.name))
+			return
+		case job := <-w.jobsChan:
+			mlog.Debug("Worker received a new candidate job.", mlog.String("worker", w.name))
+			w.doJob(&job)
+		}
+	}
+}
+
+func (w *ExtractContentWorker) Stop() {
+	mlog.Debug("Worker stopping", mlog.String("worker", w.name))
+	close(w.stopChan)
+	<-w.stoppedChan
+}
+
+func (w *ExtractContentWorker) doJob(job *model.Job) {
+	if claimed, err := w.jobServer.ClaimJob(job); err != nil {
+		mlog.Warn("Worker experienced an error while trying to claim job",
+			mlog.String("worker", w.name),
+			mlog.String("job_id", job.Id),
+			mlog.String("error", err.Error()))
+		return
+	} else if !claimed {
+		return
+	}
+
+	var err error
+	var fromTS int64 = 0
+	var toTS int64 = model.GetMillis()
+	if fromStr, ok := job.Data["from"]; ok {
+		if fromTS, err = strconv.ParseInt(fromStr, 10, 64); err != nil {
+			w.setJobError(job, model.NewAppError("ExtractContentWorker", "extrac_content.worker.do_job.invalid_input.from", nil, "", http.StatusBadRequest))
+			return
+		}
+		fromTS *= 1000
+	}
+	if toStr, ok := job.Data["to"]; ok {
+		if toTS, err = strconv.ParseInt(toStr, 10, 64); err != nil {
+			w.setJobError(job, model.NewAppError("ExtractContentWorker", "extrac_content.worker.do_job.invalid_input.to", nil, "", http.StatusBadRequest))
+			return
+		}
+		toTS *= 1000
+	}
+
+	var nFiles int
+	var nErrs int
+	for {
+		opts := model.GetFileInfosOptions{
+			Since:          fromTS,
+			SortBy:         model.FileinfoSortByCreated,
+			IncludeDeleted: false,
+		}
+		fileInfos, err := w.app.Srv().Store.FileInfo().GetWithOptions(0, 1000, &opts)
+		if err != nil {
+			w.setJobError(job, model.NewAppError("ExtractContentWorker", "extract_content.worker.do_job.file_info", nil, err.Error(), http.StatusInternalServerError))
+			return
+		}
+		if len(fileInfos) == 0 {
+			break
+		}
+		for _, fileInfo := range fileInfos {
+			if !ignoredFiles[fileInfo.Extension] {
+				mlog.Debug("extracting file", mlog.String("filename", fileInfo.Name), mlog.String("filepath", fileInfo.Path))
+				err = w.app.ExtractContentFromFileInfo(fileInfo)
+				if err != nil {
+					mlog.Warn("Failed to extract file content", mlog.Err(err), mlog.String("file_info_id", fileInfo.Id))
+					nErrs++
+				}
+				nFiles++
+			}
+		}
+		lastFileInfo := fileInfos[len(fileInfos)-1]
+		if lastFileInfo.CreateAt > toTS {
+			break
+		}
+		fromTS = lastFileInfo.CreateAt + 1
+	}
+
+	job.Data["errors"] = strconv.Itoa(nErrs)
+	job.Data["processed"] = strconv.Itoa(nFiles)
+	w.updateData(job)
+
+	mlog.Info("Worker: Job is complete", mlog.String("worker", w.name), mlog.String("job_id", job.Id))
+	w.setJobSuccess(job)
+}
+
+func (w *ExtractContentWorker) setJobSuccess(job *model.Job) {
+	if err := w.app.Srv().Jobs.SetJobSuccess(job); err != nil {
+		mlog.Error("Worker: Failed to set success for job", mlog.String("worker", w.name), mlog.String("job_id", job.Id), mlog.String("error", err.Error()))
```

Comment:
Gentle reminder to check our logging [principles](https://developers.mattermost.com/contribute/server/style-guide/#log-levels) before merging this change.